### PR TITLE
1766: Fixed warning icon overlapping text in input field.

### PR DESCRIPTION
**Pull request by adrianpastur**
# Description

Added padding and handled text overflow to avoid warning icon overlapping the text inside input fields.

# TODO
- [x] Release notes

# Related Issues / Pull Requests

https://github.com/Sage/carbon/issues/1766

# Screenshots / Gifs

In-app QA:
screen shot 2018-06-22 at 15 47 21

Carbon website QA:
screen shot 2018-07-17 at 10 28 23
screen shot 2018-07-17 at 10 28 39
screen shot 2018-07-17 at 10 29 21

# Testing Instructions

Verify input fields that will use icons such as the warning icon work as expected.

### CODE REVIEW

**Comment by andrewjtait** [comment]
ℹ️ Need to QA that the input width doesn't jump around as a result of the padding changing.

File: src/utils/decorators/input/input.scss [hunk 1]
```diff
@@ -255,6 +255,8 @@
   .common-input__input--#{$type},
   .common-input__input--#{$type}:hover {
     border-color: $color !important; // this is the correct use of !important
+    text-overflow: ellipsis;
+    padding-right: 60px;
```
Comment (ianoxley) [inline src/utils/decorators/input/input.scss]: Can you sort the CSS properties please?

**Comment by adrianpastur** [comment]
@andrewjtait @SeanArmstrong would an IE QA of the carbon website suffice?

File: src/utils/decorators/input/input.scss [hunk 2]
```diff
@@ -255,6 +255,8 @@
   .common-input__input--#{$type},
   .common-input__input--#{$type}:hover {
     border-color: $color !important; // this is the correct use of !important
+    padding-right: 60px;
```
Comment (andrewjtait) [inline src/utils/decorators/input/input.scss]: Is this not going to affect ALL inputs? Whereas you want to target just dropdowns (or inputs with a button to the right?) Date for example may need updating in a different way.

Comment (adrianpastur) [reply src/utils/decorators/input/input.scss]: Our scope was just for dropdowns. We can test how it affects other inputs and see if there are any issues, or I can target the change just for dropdowns

Comment (andrewjtait) [reply src/utils/decorators/input/input.scss]: You need to regression test other inputs at least, as this is a shared module you have changed. I suspect they will be negatively affected in this case.

File: src/components/dropdown-filter/dropdown-filter.scss [hunk 3]
```diff
@@ -41,3 +41,25 @@
     }
   }
 }
+
+@mixin common-input-text-wrapping($type, $color) {
+  .common-input__input--#{$type},
+  .common-input__input--#{$type}:hover {
+    border-color: $color !important; // this is the correct use of !important
```
Comment (andrewjtait) [inline src/components/dropdown-filter/dropdown-filter.scss]: did you not already have border-color?

File: change_log/next/input_icon_overlap.yml [hunk 4]
```diff
@@ -0,0 +1 @@
+Bug Fixes: "`Input` now warning icon will not overlap over text inside input field"
```
Comment (andrewjtait) [inline change_log/next/input_icon_overlap.yml]: You should update this to `Dropdown` rather than `Input`

**Pull request merged**